apps/ui/components/calendar/ReservationCalendarControls.tsx

```diff
@@ -638,175 +377,154 @@ const ReservationCalendarControls = <T extends Record<string, unknown>>({
     (n) => n?.endDatetime
   );
 
-  const submitButton = createReservation ? (
-    <SubmitButtonWrapper>
-      <LoginFragment
-        isActionDisabled={!isReservable}
-        apiBaseUrl={apiBaseUrl}
-        actionCallback={() => {
-          if (reservationUnit.pk != null && initialReservation != null) {
-            setStoredReservation({
-              ...initialReservation,
-              pk: null,
-              price: null,
-              reservationUnitPk: reservationUnit.pk ?? 0,
-            });
+  const submitButton =
+    mode === "create" ? (
+      <SubmitButtonWrapper>
+        <LoginFragment
+          isActionDisabled={!focusSlot.isReservable}
+          apiBaseUrl={apiBaseUrl}
+          actionCallback={() => storeReservationForLogin()}
+          componentIfAuthenticated={
+            <SubmitButton
+              type="submit"
+              disabled={!focusSlot.isReservable}
+              isLoading={reservationForm.formState.isSubmitting}
+              loadingText={t("reservationCalendar:makeReservationLoading")}
+              data-test="reservation__button--submit"
+            >
+              {t("reservationCalendar:makeReservation")}
+            </SubmitButton>
           }
-        }}
```

Comment:
Agreed - I'd be tempted to "fix it all" in this PR - but with all the changes introduced here already it might be better to do a proper splitting-into-reasonably-sized-components themed branch/PR.